app/DynamicsAdapter/DynamicsAdapter.Web/Auth/OAuthApiClient.cs

```diff
@@ -32,9 +32,17 @@ public OAuthApiClient(HttpClient httpClient, IOptionsMonitor<OAuthOptions> oAuth
 
         public async Task<Token> GetRefreshToken(CancellationToken cancellationToken)
         {
-
+            if (_httpClient.DefaultRequestHeaders.Contains("client-request-id")){
+                _httpClient.DefaultRequestHeaders.Remove("client-request-id");
+            }
             _httpClient.DefaultRequestHeaders.Add("client-request-id", Guid.NewGuid().ToString());
+            if (_httpClient.DefaultRequestHeaders.Contains("return-client-request-id")){
+                _httpClient.DefaultRequestHeaders.Remove("return-client-request-id");
```

Comment:
How about doing something like this for the rest of the entries (line 39 and below):
if (!_httpClient.DefaultRequestHeaders.Contains("return-client-request-id")){
  _httpClient.DefaultRequestHeaders.Add("return-client-request-id", "true");
}  
if (!_httpClient.DefaultRequestHeaders.Contains("Accept")){
  _httpClient.DefaultRequestHeaders.Add("Accept", "application/json");
}
or doing the following for the whole set of changes

_httpClient.DefaultRequestHeaders.clear()
_httpClient.DefaultRequestHeaders.Add("client-request-id", Guid.NewGuid().ToString());
_httpClient.DefaultRequestHeaders.Add("return-client-request-id", "true");
 _httpClient.DefaultRequestHeaders.Add("Accept", "application/json");

Then in testing we can at least assess that there are not multiple values for the same key after more than one calls. That would suffice.

Reply:
Both the options above are much riskier:
1) Checking for value and not adding if it exists is not recommended in this case.  We *always* want to add the client-request-id/guid that we have in memory as this may change.
   - It would be safe to do this for the return-client-request-id and Accept keys as those values are hardcoded and never change. 

2) Clearing all headers is dangerous and not a good options as there is potentially other headers in request such as a bearer token which would also be cleared and then not added back in.  We should only remove those headers that are being added back.

Reply:
Updated logic on static/hardcoded header values